### PR TITLE
feat(providers): support alwaysdata hosting provider

### PR DESCRIPTION
**Pull request by m4dz**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

[alwaysdata](https://www.alwaysdata.com/en/) is a French web hosting company supporting a large range of languages and frameworks such as Node.js and Nitro. This PR adds a preset and deployment documentation page dedicated to this provider.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/2.deploy/providers/alwaysdata.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+3. On your admin panel, [create a new site](https://admin.alwaysdata.com/site/add/) for your app with the following features:
+   - *Addresses*: `[account-name].alwaysdata.net`
+   - *Type*: Node.js
+   - *Command*: `node ./output/server/index.mjs`
+   - *Working directory*: `www/my-app` (adapt it to your deployment path)
+   - *Environment*:
+     ```
+     NITRO_PRESET=alwaysdata
+     ```
+   - *Node.js version*: pick at least `16.0.0` if needed, but `Default version` is fine.
```
Comment (pi0) [inline docs/content/2.deploy/providers/alwaysdata.md:32]: 18 should be at least picked. Or we can remove it if you default to 18+

File: docs/content/2.deploy/providers/alwaysdata.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
```
Comment (pi0) [inline docs/content/2.deploy/providers/alwaysdata.md:33]: Would be nice to also suggest SIGHUP configuration

File: docs/content/2.deploy/providers/alwaysdata.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+2. Get a free 100Mb plan to host your app.
+
+### Local deployment
+
+1. Build your project locally with `NITRO_PRESET=alwaysdata yarn build`
+
+2. [Upload your app](https://help.alwaysdata.com/en/remote-access/) to your account in its own directory (e.g. `$HOME/www/my-app`).
+
+3. On your admin panel, [create a new site](https://admin.alwaysdata.com/site/add/) for your app with the following features:
+   - *Addresses*: `[account-name].alwaysdata.net`
```
Comment (pi0) [inline docs/content/2.deploy/providers/alwaysdata.md:24]: It seems starting free plan setup, i need to update only account and cannot create new domain like `nitro.alwaysdata.net` it gives "domain does not belongs to you" error.

Should it be "edit exciting domain" ?

File: docs/content/2.deploy/providers/alwaysdata.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+### Pre-requisites
+
+1. [Register a new profile](https://www.alwaysdata.com/en/register/) on alwaysdata platform if you don't have one.
+
+2. Get a free 100Mb plan to host your app.
+
+### Local deployment
+
+1. Build your project locally with `NITRO_PRESET=alwaysdata yarn build`
```
Comment (pi0) [inline docs/content/2.deploy/providers/alwaysdata.md:19]:
```suggestion
1. Build your project locally with `NITRO_PRESET=alwaysdata npm run build`
```

File: src/presets/alwaysdata.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { defineNitroPreset } from "../preset";
+
+export const digitalOcean = defineNitroPreset({
```
Comment (pi0) [inline src/presets/alwaysdata.ts:3]:
```suggestion
export const alwaysdata = defineNitroPreset({
```

File: docs/content/2.deploy/providers/alwaysdata.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+2. [Upload your app](https://help.alwaysdata.com/en/remote-access/) to your account in its own directory (e.g. `$HOME/www/my-app`).
+
+3. On your admin panel, [create a new site](https://admin.alwaysdata.com/site/add/) for your app with the following features:
+   - *Addresses*: `[account-name].alwaysdata.net`
+   - *Type*: Node.js
+   - *Command*: `node ./output/server/index.mjs`
```
Comment (pi0) [inline docs/content/2.deploy/providers/alwaysdata.md:26]: There is no instructions how to upload website but it is `.output` by default i expect directories inside it (public/server) being copied to the hosting

**Comment by pi0** [comment]
Thanks for pull request dear @m4dz. I have made it WIP until we can solve all issues and have a successful deployment from the instructions.

File: docs/content/2.deploy/providers/alwaysdata.md (hunk 6 above)
Comment (m4dz) [reply docs/content/2.deploy/providers/alwaysdata.md:28]: Correct, I added a mention in the 2. above to add clarity on how deploying, for example using SSH

File: docs/content/2.deploy/providers/alwaysdata.md (hunk 3 above)
Comment (m4dz) [reply docs/content/2.deploy/providers/alwaysdata.md]: Right. It's not limited to the free tier: all plans at alwaysdata come with a default URL in the form of `[account_name].alwaysdata.net`. But you can create as many accounts as you want, and you can also link your own domains (or register new ones). So unless your account name is `nitro`, you can't use it as your URL.
I've added a section in the `Create account` pre-requisites to make it clearer. Does it sound good?

File: src/presets/alwaysdata.ts (hunk 5 above)
Comment (m4dz) [reply src/presets/alwaysdata.ts]: My bad 😰

**Comment by m4dz** [comment]
Thanks @pi0 for the fast review! I made some changes, please tell me if it sounds good now :+1:

File: docs/content/2.deploy/providers/alwaysdata.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+2. Get a free 100Mb plan to host your app.
+
+*Note:* Keep in mind your *account name* will be used to provide you a default URL in the form of `account_name.alwaysdata.net`, so choose it wisely. You can also link your existing domains to your account later or register as many accounts under your profile as you need.
+
+### Local deployment
+
+1. Build your project locally with `NITRO_PRESET=alwaysdata npm run build`
+
+2. [Upload your app](https://help.alwaysdata.com/en/remote-access/) to your account in its own directory (e.g. `$HOME/www/my-app`). You can use any protocol you prefer (SSH/FTP/WebDAV…); just be careful to transfer the built `.output` directory and its subdirectories. E.g. `scp -r my-app [account_name]@ssh-[account_name].alwaysdata.net:~/www`.
```
Comment (pi0) [inline docs/content/2.deploy/providers/alwaysdata.md:23]: For this in custom preset you made we can use `commands: { deploy: `scp -r ./ [account].alwaysdaa.net:~/www` }` so we show users command in the end.

Comment (m4dz) [reply docs/content/2.deploy/providers/alwaysdata.md]: Oh sweet! Thanks!

**Comment by m4dz** [comment]
Added `commands.deploy` directive, should be better now :)

File: src/presets/alwaysdata.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import { defineNitroPreset } from "../preset";
+
+export const alwaysdata = defineNitroPreset({
+  extends: "node-server",
+  commands : {
+    deploy: "scp -r . [account].alwaysdata.net:~/www/"
```
Comment (pi0) [inline src/presets/alwaysdata.ts:6]: (you can try locally, with `./` nitro smartly generated actual output directory name!)

```suggestion
    deploy: "scp -r ./ [account].alwaysdata.net:~/www/"
```

Comment (m4dz) [reply src/presets/alwaysdata.ts]: Ok, didn't notice that, my bad. I moved it to using rsync for more clarity and better compatibility.

**Review by pi0 — APPROVED** [review]
sorry got delayed

**Comment by m4dz** [comment]
No worries, thanks a lot for the merge!

**Pull request merged**